### PR TITLE
Upgrade terraform-provider-alicloud to v1.196.0

### LINKED ISSUE #315: Upgrade terraform-provider-alicloud to v1.186.0
Release details: https://github.com/aliyun/terraform-provider-alicloud/releases/tag/v1.186.0

### LINKED ISSUE #316: Upgrade terraform-provider-alicloud to v1.187.0
Release details: https://github.com/aliyun/terraform-provider-alicloud/releases/tag/v1.187.0

### LINKED ISSUE #319: Upgrade terraform-provider-alicloud to v1.188.0
Release details: https://github.com/aliyun/terraform-provider-alicloud/releases/tag/v1.188.0

### LINKED ISSUE #320: Upgrade terraform-provider-alicloud to v1.189.0
Release details: https://github.com/aliyun/terraform-provider-alicloud/releases/tag/v1.189.0

### LINKED ISSUE #321: Upgrade terraform-provider-alicloud to v1.190.0
Release details: https://github.com/aliyun/terraform-provider-alicloud/releases/tag/v1.190.0

### LINKED ISSUE #322: Upgrade terraform-provider-alicloud to v1.191.0
Release details: https://github.com/aliyun/terraform-provider-alicloud/releases/tag/v1.191.0

### LINKED ISSUE #323: Upgrade terraform-provider-alicloud to v1.192.0
Release details: https://github.com/aliyun/terraform-provider-alicloud/releases/tag/v1.192.0

### LINKED ISSUE #327: Upgrade terraform-provider-alicloud to v1.193.0
Release details: https://github.com/aliyun/terraform-provider-alicloud/releases/tag/v1.193.0

### LINKED ISSUE #328: Upgrade terraform-provider-alicloud to v1.193.1
Release details: https://github.com/aliyun/terraform-provider-alicloud/releases/tag/v1.193.1

### LINKED ISSUE #329: Upgrade terraform-provider-alicloud to v1.194.0
Release details: https://github.com/aliyun/terraform-provider-alicloud/releases/tag/v1.194.0

### LINKED ISSUE #330: Upgrade terraform-provider-alicloud to v1.194.1
Release details: https://github.com/aliyun/terraform-provider-alicloud/releases/tag/v1.194.1

### LINKED ISSUE #331: Upgrade terraform-provider-alicloud to v1.195.0
Release details: https://github.com/aliyun/terraform-provider-alicloud/releases/tag/v1.195.0

### LINKED ISSUE #332: Upgrade terraform-provider-alicloud to v1.196.0
Release details: https://github.com/aliyun/terraform-provider-alicloud/releases/tag/v1.196.0

### PR DESCRIPTION
**Pull request by iwahbe**
Fixes #332 
Fixes #331 
Fixes #330 
Fixes #329 
Fixes #328 
Fixes #327 
Fixes #323 
Fixes #322 
Fixes #321 
Fixes #320 
Fixes #319 
Fixes #316 
Fixes #315

### CODE REVIEW

File: provider/resources.go [hunk 1]
```diff
@@ -132,6 +140,7 @@ const (
 	slbMod                 = "Slb"
 	smsMod                 = "Sms"
 	tagMod                 = "Tag"
+	threatDetectionMod     = "threatDetection"
```
Comment (jazzyfresh) [inline provider/resources.go:143]:
```suggestion
	threatDetectionMod     = "ThreatDetection"
```

Comment (iwahbe) [reply provider/resources.go:143]: Good catch!

File: provider/resources.go [hunk 2]
```diff
@@ -1365,19 +1494,30 @@ func Provider() tfbridge.ProviderInfo {
...
-			"alicloud_cen_private_zones":               {Tok: dataSource(cenMod, "getPrivateZones")},
-			"alicloud_cen_instance_attachments":        {Tok: dataSource(cenMod, "getInstanceAttachments")},
-			"alicloud_cen_vbr_health_checks":           {Tok: dataSource(cenMod, "getVbrHealthChecks")},
-			"alicloud_cen_route_services":              {Tok: dataSource(cenMod, "getRouteServices")},
-			"alicloud_cen_transit_routers":             {Tok: dataSource(cenMod, "getTransitRouters")},
-			"alicloud_cen_transit_router_route_tables": {Tok: dataSource(cenMod, "getTransitRouterRouteTables")},
+			"alicloud_cen_bandwidth_limits":                             {Tok: dataSource(cenMod, "getBandwidthLimits")},
+			"alicloud_cen_bandwidth_packages":                           {Tok: dataSource(cenMod, "getBandwidthPackages")},
+			"alicloud_cen_child_instance_route_entry_to_attachments":    {Tok: dataSource(cenMod, "getChildInstanceRouteEntryToAttachments")},
+			"alicloud_cen_flowlogs":                                     {Tok: dataSource(cenMod, "getFlowlogs")},
```
Comment (jazzyfresh) [inline provider/resources.go:1500]: not sure if this matters but above you have `FlowLogs` camelcased, so either here or line #379 should be changed for consistency

File: provider/resources.go [hunk 3]
```diff
@@ -1476,19 +1622,20 @@ func Provider() tfbridge.ProviderInfo {
...
 			"alicloud_cr_ee_sync_rules":                  {Tok: dataSource(csMod, "getRegistryEnterpriseSyncRules")},
-			"alicloud_ack_service":                       {Tok: dataSource(csMod, "getAckService")},
-			"alicloud_cs_kubernetes_permissions":         {Tok: dataSource(csMod, "getKubernetesPermission")},
-			"alicloud_cs_kubernetes_addons":              {Tok: dataSource(csMod, "getKubernetesAddons")},
+			"alicloud_cs_cluster_credential":             {Tok: dataSource(csMod, "getClusterCredential")},
+			"alicloud_cs_edge_kubernetes_clusters":       {Tok: dataSource(csMod, "getEdgeKubernetesClusters")},
 			"alicloud_cs_kubernetes_addon_metadata":      {Tok: dataSource(csMod, "getKubernetesAddonMetadata")},
+			"alicloud_cs_kubernetes_addons":              {Tok: dataSource(csMod, "getKubernetesAddons")},
+			"alicloud_cs_kubernetes_clusters":            {Tok: dataSource(csMod, "getKubernetesClusters")},
+			"alicloud_cs_kubernetes_permissions":         {Tok: dataSource(csMod, "getKubernetesPermission")},
```
Comment (jazzyfresh) [inline provider/resources.go:1635]:
```suggestion
			"alicloud_cs_kubernetes_permissions":         {Tok: dataSource(csMod, "getKubernetesPermissions")},
```

**Review by jazzyfresh — APPROVED** [review]
Reviewed `go.mod` and `resources.go`, LGTM. 

Let me know if you need any additional codegened files reviewed.

File: provider/resources.go (hunk 2 above)
Comment (iwahbe) [reply provider/resources.go:1500]: The version from L379 predates this PR, so I'll change L1500.

File: provider/resources.go (hunk 3 above)
Comment (iwahbe) [reply provider/resources.go:1635]: Since this was already in the provider, I'm going to leave it rather then add an alias to fix it.

**Comment by t0yv0** [comment]
Spot-checked Resource "alicloud:gpdb/instance:Instance" missing input "instanceClass" - I became "instanceSpec" - looks like a fine small change made upstream.

However we seem to have a blocker in the code generation for Go:

```
package github.com/pulumi/pulumi-alicloud/sdk/v3/go/alicloud/cen: case-insensitive file name collision: "getFlowLogs.go" and "getFlowlogs.go"
```

**Comment by iwahbe** [comment]
> Spot-checked Resource "alicloud:gpdb/instance:Instance" missing input "instanceClass" - I became "instanceSpec" - looks like a fine small change made upstream.
> 
> However we seem to have a blocker in the code generation for Go:
> 
> ```
> package github.com/pulumi/pulumi-alicloud/sdk/v3/go/alicloud/cen: case-insensitive file name collision: "getFlowLogs.go" and "getFlowlogs.go"
> ```

That's because I changed the capitalization of a name, and we don't delete the old SDKs before we regenerate the new ones. I'm fixing regenerating the SDK.

**Pull request merged**